mods/ra/maps/allies-08a/allies08a-AI.lua

```diff
@@ -0,0 +1,239 @@
+	Upgraded = { {"4tnk", "3tnk", "ftrk", "apc", "apc", "e1", "e1", "e1", "e1", "e1", "e2", "e2", "e2" }, { "3tnk", "v2rl", "ftrk", "apc", "apc", "e1", "e1", "e1", "e1", "e1", "e4", "e4", "e4" } }
+}
+GroundAttackPaths = 
+{ 
+	{ SovEntry1.Location, ParaLZ3.Location, AttackChrono.Location },
+	{ SovEntry2.Location, ParaLZ5.Location, AttackChrono.Location },
+	{ SovEntry3.Location, ParaLZ5.Location, AttackChrono.Location },
+	{ MCVEntry.Location, ParaLZ2.Location, AttackChrono.Location }
+}
+GroundWavesDelays =
+{
+	easy = 4,
+	normal = 3,
+	hard = 2
+}
+
+WTransUnits = { { "4tnk", "3tnk", "e2", "e2", "e2" }, { "3tnk", "v2rl", "e4", "e4", "e4" } }
+WTransWays =
+{
+	{ DDEntry.Location, WaterLanding1.Location },
+	{ WaterEntry2.Location, WaterLanding2.Location },
+	{ WaterEntry2.Location, WaterLanding3.Location }
+}
+WTransDelays =
+{
+	easy = 5,
+	normal = 4,
+	hard = 3
+}
+
+ParadropLZs = { ParaLZ1.CenterPosition, ParaLZ2.CenterPosition, ParaLZ3.CenterPosition, ParaLZ4.CenterPosition, ParaLZ5.CenterPosition }
+ParadropDelays =
+{
+	easy = 3,
+	normal = 2,
+	hard = 1
+}
+
+BombDelays =
+{
+	easy = 4,
+	normal = 3,
+	hard = 2
+}
+
+ProductionInterval =
+{
+	easy = DateTime.Seconds(20),
+	normal = DateTime.Seconds(10),
+	hard = DateTime.Seconds(5)
+}
+
+SendAttackGroup = function()
+	if #AttackGroup < AttackGroupSize then
+		return
+	end
+
+	Utils.Do(AttackGroup, function(unit)
+		if not unit.IsDead then
+			Trigger.OnIdle(unit, unit.Hunt)
+		end
+	end)
+
+	AttackGroup = { }
+end
+
+ProduceInfantry = function()
+	if USSRRax.IsDead or USSRRax.Owner ~= ussr then
+		return
+	end
+
+	ussr.Build({ Utils.Random(SovietInfantry) }, function(units)
+		table.insert(AttackGroup, units[1])
+		SendAttackGroup()
+		Trigger.AfterDelay(ProductionInterval[Map.LobbyOption("difficulty")], ProduceInfantry)
+	end)
+end
+
+ProduceVehicles = function()
+	if USSRWarFactory.IsDead or USSRWarFactory.Owner ~= ussr then
+		return
+	end
+
+	ussr.Build({ Utils.Random(SovietVehicles) }, function(units)
+		table.insert(AttackGroup, units[1])
+		SendAttackGroup()
+		Trigger.AfterDelay(ProductionInterval[Map.LobbyOption("difficulty")], ProduceVehicles)
+	end)
+end
+
+ProduceAircraft = function()
+	if (Airfield1.IsDead or Airfield1.Owner ~= ussr) and (Airfield2.IsDead or Airfield2.Owner ~= ussr) then
+		return
+	end
+
+	ussr.Build(SovietAircraftType, function(units)
+		local mig = units[1]
+		Migs[#Migs + 1] = mig
+
+		Trigger.OnKilled(mig, ProduceAircraft)
+
+		local alive = Utils.Where(Migs, function(y) return not y.IsDead end)
+		if #alive < 2 then
+			Trigger.AfterDelay(DateTime.Seconds(ProductionInterval[Map.LobbyOption("difficulty")] / 2), ProduceAircraft)
+		end
+
+		TargetAndAttack(mig)
+	end)
+end
+
+TargetAndAttack = function(mig, target)
+	if not target or target.IsDead or (not target.IsInWorld) then
+		local enemies = Utils.Where(greece.GetActors(), function(actor)
+			return
+				actor.HasProperty("Health") and
```

Comment:
The format looks a bit awkward, imho. Would you mind moving everything on one line again?
`return actor.HasProperty("Health") and actor.Type ~= "brik"`